src/protocols/secure_channel/NetworkProvisioning.cpp

```diff
@@ -156,38 +156,32 @@ size_t NetworkProvisioning::EncodedStringSize(const char * str)
 
 CHIP_ERROR NetworkProvisioning::EncodeString(const char * str, Encoding::LittleEndian::BufferWriter & bbuf)
 {
-    CHIP_ERROR err  = CHIP_NO_ERROR;
-    size_t length   = strlen(str);
-    uint16_t u16len = static_cast<uint16_t>(length);
-    VerifyOrExit(CanCastTo<uint16_t>(length), err = CHIP_ERROR_INVALID_ARGUMENT);
+    const size_t length   = strlen(str);
+    const uint16_t u16len = static_cast<uint16_t>(length);
+    VerifyOrReturnError(CanCastTo<uint16_t>(length), CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
The order of these two lines should be switched, now that we can do it without the compiler complaining at us.  So do the check first, then the line with the cast.
```suggestion
    VerifyOrReturnError(CanCastTo<uint16_t>(length), CHIP_ERROR_INVALID_ARGUMENT);
    const uint16_t u16len = static_cast<uint16_t>(length);
```